net-misc/pingu/pingu-1.5.ebuild

```diff
@@ -0,0 +1,60 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit systemd tmpfiles toolchain-funcs
+
+DESCRIPTION="Policy routing daemon with failover and load-balancing"
+HOMEPAGE="https://github.com/ncopa/pingu"
+SRC_URI="https://github.com/ncopa/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug doc"
+
+RDEPEND="dev-libs/libev"
+DEPEND="${RDEPEND}
+	sys-kernel/linux-headers
+	virtual/pkgconfig
+	doc? ( app-text/asciidoc )"
+
+src_prepare() {
+	default
+
+	# Fix compilation issue
```

Comment:
Is it fixed upstream? Link to upstream commit would be helpful, or some explanation **why** you're doing stuff like that.

Reply:
It is not, upstream is dead for almost 3 years, but the code is working. I could create a PR into his repo.
Without this, it fails to build.

Reply:
Is it working? How do you know? This really looks the kind of 'it failed to build, so I just removed some line I didn't understand and it built now'.

Reply:
> Is it working?

Yes, it is. I have no interest in submitting a non-working piece of code.

> How do you know?

1. Because I have 1.5 years of production experience with this tool.
2. Because this is second attempt to submit it — the first [one](https://github.com/gentoo/gentoo/pull/6916) was at Jan 18, and I had a confidence in this package then. It didn't decrease six months later.
3. Because I tried to manually build it before:
```
cc -march=native -O2 -pipe -mmmx -msse4.2 -mssse3 -mfpmath=sse -fomit-frame-pointer -Wall -DHAVE_CONFIG_H -I../ -DPINGU_VERSION=\"\" -Wall -Wstrict-prototypes -D_GNU_SOURCE -std=gnu99 -DDEFAULT_PIDFILE=\"/run/pingu/pingu.pid\" -DDEFAULT_CONFIG=\"/usr/etc/pingu/pingu.conf\" -DDEFAULT_ADM_client=\"/run/pingu/pingu.ctl\"  -c icmp.c
icmp.c: In function 'icmp_send_frag_needed':
icmp.c:261:14: error: 'struct <anonymous>' has no member named '__unused'
  icp->un.frag.__unused = 0;
              ^
make: *** [Makefile:82: icmp.o] Error 1
```
and looked over the code what this struct for.
And I think that my fix is correct.

But if you would like to be absolutely sure, I have created a [PR](https://github.com/ncopa/pingu/pull/4) and we just could wait for upstream decision if he is interested in it.

Reply:
> Link to upstream commit would be helpful

https://github.com/ncopa/pingu/commit/8c9b53cf3ecd1ae33291a1a5640b7b60e85f2601